metrics/metrics.go

```diff
@@ -163,11 +161,15 @@ func CollectProcessMetrics(refresh time.Duration) {
 		diskWriteBytesCounter = GetOrRegisterCounter("system/disk/writebytes", DefaultRegistry)
 	)
 
+	// Avoid divide-by-zero the first time through the loop.
+	lastCollectionTime := time.Now().Add(-refresh)
+
 	// Iterate loading the different stats and updating the meters.
 	now, prev := 0, 1
 	for ; ; now, prev = prev, now {
 		// CPU
 		ReadCPUStats(&cpustats[now])
+		refreshFreq := int64(time.Since(lastCollectionTime) / time.Second)
```

Comment:
As you said, the time interval might be 3.3s, but the extra 0.3 is truncated here.

Reply:
Good point!  I propose changing to the code below.  I should say, I am very new to Go, so if there is a better way, please give me a hint!  🙏  If it looks good, I will push a new commit.

``` Go
// CPU
ReadCPUStats(&cpustats[now])
refreshFreq := time.Since(lastCollectionTime).Seconds()
cpuSysLoad.Update(int64(float64(cpustats[now].GlobalTime - cpustats[prev].GlobalTime) / refreshFreq))
cpuSysWait.Update(int64(float64(cpustats[now].GlobalWait - cpustats[prev].GlobalWait) / refreshFreq))
cpuProcLoad.Update(int64(float64(cpustats[now].LocalTime - cpustats[prev].LocalTime) / refreshFreq))
```

Reply:
See new commit with above changes.